src/Integrations/Integrations/PHPRedis/PHPRedisIntegration.php

```diff
@@ -55,6 +55,29 @@ public function init()
         \DDTrace\trace_method('Redis', 'open', $traceConnectOpen);
         \DDTrace\trace_method('Redis', 'popen', $traceConnectOpen);
 
+        $traceNewCluster = function (SpanData $span, $args) {
+            if (isset($args[0]) && \is_string($args[0])) {
+                $hostOrUDS = $args[0];
+            } else {
+                $hostOrUDS = (isset($args[1]) && \is_array($args[1]) && !empty($args[1])) ? $args[1][0] : '127.0.0.1';
+            }
+            $url = parse_url($hostOrUDS);
+            $span->meta[Tag::TARGET_HOST] = (is_array($url) && isset($url["host"])) ? $url["host"] : '127.0.0.1';
+            $span->meta[Tag::TARGET_PORT] = (is_array($url) && isset($url["port"])) ? $url["port"] : 6379;
```

Comment:
This does not make much sense with the $name arg ($args[0]) being non-null.
I think what makes most sense here is extracting the first seed from `redis.clusters.seeds` then. the host/port meta tags ought to be true hosts or ports, not config names.

Reply:
As per RedisCluster documentation: https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#loading-a-cluster-configuration-by-name

you can define clusters in the ini file and reference them by name, this logic allows compatibility with this scenario.

Reply:
Yes, I'm aware, let me expand a bit more on my comment:

Given a `new RedisCluster('test')`, `$hostOrUDS` will now contain `test`. Subsequently `meta[Tag::TARGET_HOST]` contains now `test` and `meta[Tag::TARGET_PORT]` is `6379`.

However, this is just a name, not a true hostname+port pair (which is required for TARGET_HOST and TARGET_PORT). Hence my suggestion is to go to `redis.clusters.seeds` and extract the first encountered host for the given name.

e.g. given `redis.clusters.seeds = "test[]=localhost:7000&test[]=localhost:7001"`

We would assign TARGET_HOST => localhost, TARGET_PORT => 7000.

The service name should contain the name if one is given, otherwise the host/port/uds. (just like you implemented it here)

Reply:
I got your point, I'll have the changes in few minutes